iroh-gateway/src/metrics.rs

```diff
@@ -23,3 +26,12 @@ pub fn register_counters() {
         "Total number of requests received by the gateway"
     );
 }
+
+pub fn get_current_trace_id() -> String {
```

Comment:
this should return the actual traceid and the stringification should happen when used